backend/backendutil/search.go

```diff
@@ -86,7 +86,11 @@ func Match(e *message.Entity, seqNum, uid uint32, date time.Time, flags []string
 				ok := false
 				values := e.Header.FieldsByKey(key)
 				for values.Next() {
-					if matchString(values.Value(), wantValue) {
+					decoded, err := values.Text()
+					if err != nil {
```

Comment:
On error (e.g. unknown charset) the raw value is returned. Maybe we should still `matchString` it?

Reply:
so
```go
decoded, _ := values.Text()
if matchString(decoded, wantValue) {
  ok = true
  break
}
```

I thought it does not return a meaningful value if error is not nil.

Reply:
Yeah. I thought I documented it, but it was probably for body charset errors.